### DISCUSSION TITLE
Automating hybrid app for MacOs with Appium2 and WebriverIO

### DISCUSSION THREAD

**Discussion by arturgorsky** [appium/appium#19271] [General]
Hi, I wanted to automate hybrid app that uses webView to embed content into desktop MacOS app. I'm using webdriverIO. The native components automation works fine, but the webView components are missing most of the attributes (f.e. the button  XCUIElementTypeButton elementType="9" identifier="" value="" label="" title="" placeholderValue="" enabled="true" selected="false" x="1580" y="542" width="18" height="21") is missing most of the properties and only the aria-label gets passed as label.

I use the following capabilities:
```
capabilities: [
    {
      platformName: "Mac",
      "appium:automationName": "Mac2",
      "appium:bundleId": "com.appName",
    },
  ],
```

I've searched for more information, but I found only some settings for appium1.x. Any tips will be helpful

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium-mac2-driver does not provide WebView context support.

> only the aria-label gets passed as label.

This was because I guess iOS's accessibility framework parsed them as native context, then the XCTest framework was able to identify them.

**Reply by arturgorsky** [reply]
Seems like it, I tried switching the context to 'WEBVIEW' but it says that the session is not active. I assume I need to have Chromedriver running, but I am not sure how to run Appium Mac and Chromedriver simultaneously

**Reply by KazuCocoa** [reply]
Perhaps you might need to manage two driver instances, one is mac2 and the other is chromedriver's one. For chromedriver's one, you could use https://github.com/appium/appium-chromium-driver or vanilla chromedriver.
